python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -1187,21 +1206,30 @@ def fetch_column_metadata(
                 A set of corresponding Dagster events for dbt models, with column metadata attached,
                 yielded in the order they are emitted by dbt.
         """
-        fetch_metadata = lambda invocation, event: _fetch_column_metadata(
-            invocation, event, with_column_lineage
+        fetch_metadata = lambda invocation, ctx: _fetch_column_metadata(
+            invocation, ctx, with_column_lineage
         )
-        return self._attach_metadata(fetch_metadata)
+        return self.attach_metadata(fetch_metadata)
 
-    def _attach_metadata(
+    @public
+    @experimental
```

Comment:
Should make sure this doesn't raise undesired experimental warnings when we invoke it internally.